akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -2854,9 +2854,33 @@ final class Flow[In, Out, Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends Gr
    * by the monitor unchanged. Note that the monitor inserts a memory barrier every time it processes an
    * event, and may therefor affect performance.
    * The `combine` function is used to combine the `FlowMonitor` with this flow's materialized value.
+   *
+   * @deprecated Use monitor() or monitorMat(combine) instead
    */
+  @Deprecated
+  @deprecated("2.5.12", "Use monitor() or monitorMat(combine) instead")
   def monitor[M]()(combine: function.Function2[Mat, FlowMonitor[Out], M]): javadsl.Flow[In, Out, M] =
-    new Flow(delegate.monitor()(combinerToScala(combine)))
+    new Flow(delegate.monitorMat(combinerToScala(combine)))
+
+  /**
+   * Materializes to `FlowMonitor[Out]` that allows monitoring of the current flow. All events are propagated
+   * by the monitor unchanged. Note that the monitor inserts a memory barrier every time it processes an
+   * event, and may therefor affect performance.
+   * The `combine` function is used to combine the `FlowMonitor` with this flow's materialized value.
+   *
+   * @deprecated Use monitor() or monitorMat(combine) instead
+   */
+  def monitorMat[M](combine: function.Function2[Mat, FlowMonitor[Out], M]): javadsl.Flow[In, Out, M] =
+    new Flow(delegate.monitorMat(combinerToScala(combine)))
+
+  /**
+   * Materializes to `FlowMonitor[Out]` that allows monitoring of the current flow. All events are propagated
+   * by the monitor unchanged. Note that the monitor inserts a memory barrier every time it processes an
+   * event, and may therefor affect performance.
+   * The `combine` function is used to combine the `FlowMonitor` with this flow's materialized value.
+   */
+  def monitor(): Flow[In, Out, FlowMonitor[Out]] =
+    monitorMat(Keep.right)
```

Comment:
Open question here about which keep to use as default. We normally use `Keep.left`, but since the entire purpose of monitor is to add this mat value, the default being keep.right makes sense... But we'd lose the left and do something differently than all of the other ops -- should it rather be `Keep.both` perhaps then?

Reply:
I think Right is good here, for both there is monitorMat

Reply:
Good question. Do we have any other such exceptions to the keep.left rule. I think it can become a mess if we don't follow a single rule. Perhaps that was the reason why the original `monitor` had the `combine`, it's not useful without it?

Reply:
I think this is a fair place to step away from the rule, given that it does not make sense to introduce a monitor mat val and then throw it away. I don't think that clouds the general rule.

Reply:
Then it should be `Both` and clearly documented.
I'm not super happy about breaking the rule - _sounds like I have become the German guy_.

Reply:
I’d be fine with Both and clearly documenting, it makes sense here. Will change to that